runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientCreateTopicsFactory.java

```diff
@@ -0,0 +1,1676 @@
+                                                                                t.item(i -> i
+                                                                                    .name(ts.name)
+                                                                                    .error(ts.error)
+                                                                                    .message(ts.message))))))
+                                                            .build()
+                                                            .sizeof()));
+            }
+        }
+
+
+        private void doApplicationEnd(
+            long traceId)
+        {
+            state = KafkaState.closedReply(state);
+            doEnd(application, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                    traceId, client.authorization, EMPTY_EXTENSION);
+        }
+
+        private void doApplicationAbort(
+            long traceId)
+        {
+            state = KafkaState.closedReply(state);
+            doAbort(application, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                    traceId, client.authorization, EMPTY_EXTENSION);
+        }
+
+        private void doApplicationWindow(
+            long traceId,
+            long budgetId,
+            int minInitialNoAck,
+            int minInitialPad,
+            int minInitialMax)
+        {
+            final long newInitialAck = Math.max(initialSeq - minInitialNoAck, initialAck);
+
+            if (newInitialAck > initialAck || minInitialMax > initialMax || !KafkaState.initialOpened(state))
+            {
+                initialAck = newInitialAck;
+                assert initialAck <= initialSeq;
+
+                initialMax = minInitialMax;
+
+                state = KafkaState.openedInitial(state);
+
+                doWindow(application, originId, routedId, initialId, initialSeq, initialAck, initialMax,
+                        traceId, client.authorization, budgetId, minInitialPad);
+            }
+        }
+
+        private void doApplicationReset(
+            long traceId,
+            Flyweight extension)
+        {
+            state = KafkaState.closedInitial(state);
+
+            doReset(application, originId, routedId, initialId, initialSeq, initialAck, initialMax,
+                    traceId, client.authorization, extension);
+        }
+
+        private void doApplicationAbortIfNecessary(
+            long traceId)
+        {
+            if (KafkaState.replyOpening(state) && !KafkaState.replyClosed(state))
+            {
+                doApplicationAbort(traceId);
+            }
+        }
+
+        private void doApplicationResetIfNecessary(
+            long traceId,
+            Flyweight extension)
+        {
+            if (KafkaState.initialOpening(state) && !KafkaState.initialClosed(state))
+            {
+                doApplicationReset(traceId, extension);
+            }
+        }
```

Comment:
Please avoid creating separate `IfNecessary` methods and put check directly in the original method instead.